packages/nextjs/components/SwitchTheme.tsx

```diff
@@ -0,0 +1,32 @@
+//SwitchTheme.tsx
+import React, { useEffect } from "react";
+import { useDarkMode, useIsMounted } from "usehooks-ts";
+import { MoonIcon, SunIcon } from "@heroicons/react/24/outline";
+const SwitchTheme = () => {
+  const { isDarkMode, toggle } = useDarkMode(false);
+  const isMounted = useIsMounted();
+
+  useEffect(() => {
+    const body = document.body;
+    body.setAttribute("data-theme", isDarkMode ? "scaffoldEthDark" : "scaffoldEth");
+  }, [isDarkMode]);
+
+  return (
+    <div className="flex space-x-2">
+      <input
+        id="theme-toggle"
+        type="checkbox"
+        className="toggle toggle-primary bg-primary"
+        onChange={toggle}
+        checked={isDarkMode}
+      />
+      {isMounted() && (
```

Comment:
The reason for doing this is there is the inconsistent output we don't do it : 

**Case when there is inconsistent output**  : 
- select dark mode and close the browser
- now when you open it again since during the build of NEXT.js page localstorage is not available initially `isDarkMode` is `false` (in theory it should have been true because we choose dark mode earlier)

demo of inconsistent behavior(withou `isMounted()`): 

https://user-images.githubusercontent.com/80153681/210214863-5532bb17-0cbc-4799-ba33-1525e249591e.mov

Take a look at `Icons` ^

Reply:
This is a good catch!

The `useIsMounted` could be useful for nextjs hydration errors!